src/components/inputAndLabel/InputAndLabel.styles.ts

```diff
@@ -10,24 +10,24 @@ export const InputFieldStyled = styled.input<{
   padding: 10px;
   width: 100%;
   height: ${({ $height }) => $height};
-  background-color: var(--cream);
-  color: var(--primary-green);
+  background-color: ${({ theme }) => theme.colors.cream};
+  color: ${({ theme }) => theme.colors.primaryGreen};
 
   &::placeholder {
     opacity: 30%;
-    color: var(--primary-green);
-    font-family: var(--primary-font);
-    font-size: 14px;
-    font-weight: 400;
-    line-height: 1.2;
-    letter-spacing: 0.6px;
+    color: ${({ theme }) => theme.colors.primaryGreen};
+    font-family: ${({ theme }) => theme.fonts.primaryFont};
+    font-size: ${({ theme }) => theme.fontSizes.smallParagraph}px;
+    font-weight: ${({ theme }) => theme.fontWeights.normal};
+    line-height: ${({ theme }) => theme.lineHeight.smallParagraph};
+    letter-spacing: ${({ theme }) => theme.letterSpacing.smallParagraph}px;
   }
```

Comment:
that's the issue with typography I mentioned some time ago - it's tedious and hard to maintain without predefined variants. Placeholder is an exception (you won't use a component for it) but the problem is visible everywhere else in this PR where you defined font-sizes and all the remaining font-related styles. 

Please try to find some info on how this can be handled with styled components and let me know what you found. We can implement it in the next PR if you agree it's useful (or you can convince me it's not a necessity right now) :)

Reply:
as agreed during our call, I'm creating Text.tsx component for basic typography